lib/cluster.js

```diff
@@ -591,6 +633,16 @@ function workerInit() {
     send(message, function(reply, handle) {
       if (obj._setServerData) obj._setServerData(reply.data);
 
+      if (+options.port === 0 && reply.sockname) {
```

Comment:
Nit: As the `+options.port` coercion is done twice you might just move that in a variable above?